mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+        if (ar.succeeded()) {
+          LOGGER.info("Subscribed to topics [{}]", StringUtils.join(topics, ","));
+          future.complete(true);
+        } else {
+          LOGGER.error("Could not subscribe to some of the topics [{}]", ar.cause(), StringUtils.join(topics, ","));
+          future.fail(ar.cause());
+        }
+      }).handler(getEventReceivedHandler(tenantId, okapiHeaders));
+    return future;
+  }
+
+  private Handler<KafkaConsumerRecord<String, String>> getEventReceivedHandler(String tenantId, Map<String, String> okapiHeaders) {
+    return record -> {
+      try {
+        String value = record.value();
+        LOGGER.info("Received event {}", value);
+        Event event = new JsonObject(value).mapTo(Event.class);
+        saveAuditMessage(event, tenantId, AuditMessage.State.RECEIVED);
+        deliverEvent(event, tenantId, okapiHeaders);
+      } catch (Exception e) {
+        LOGGER.error("Error reading event value", e);
+      }
+    };
+  }
+
+  private void deliverEvent(Event event, String tenantId, Map<String, String> okapiHeaders) {
+    messagingModuleDao.get(new MessagingModuleFilter()
+      .withTenantId(tenantId)
+      .withModuleRole(SUBSCRIBER)
+      .withEventType(event.getEventType()))
+      .compose(messagingModuleList -> {
+        if (CollectionUtils.isEmpty(messagingModuleList)) {
+          String errorMessage = format("There is no SUBSCRIBERS registered for event type %s. Event %s will not be delivered", event.getEventType(), event.getId());
+          LOGGER.error(errorMessage);
+        } else {
+          messagingModuleList
+            .forEach(subscriber -> doRequest(event, subscriber.getSubscriberCallback(), okapiHeaders)
+              .setHandler(getEventDeliveredHandler(event, tenantId, subscriber)));
+        }
+        return Future.succeededFuture();
+      });
+  }
+
+  private Handler<AsyncResult<HttpClientResponse>> getEventDeliveredHandler(Event event, String tenantId, MessagingModule subscriber) {
+    return ar -> {
+      if (ar.failed()) {
+        LOGGER.error("Event {} was not delivered to {}", ar.cause(), event.getId(), subscriber.getSubscriberCallback());
+        saveAuditMessage(event, tenantId, AuditMessage.State.REJECTED);
+      } else if (ar.result().statusCode() != 200 && ar.result().statusCode() != 201 && ar.result().statusCode() != 204) {
+        LOGGER.error("Error delivering event {} to {}, response status code is {}, {}",
+          event.getId(), subscriber.getSubscriberCallback(), ar.result().statusCode(), ar.result().statusMessage());
+        saveAuditMessage(event, tenantId, AuditMessage.State.REJECTED);
+      } else {
+        LOGGER.info("Delivered event {} to {}", event.getId(), subscriber.getSubscriberCallback());
+        saveAuditMessage(event, tenantId, AuditMessage.State.DELIVERED);
+      }
+    };
+  }
+
+  private Future<HttpClientResponse> doRequest(Event event, String callbackPath, Map<String, String> okapiHeaders) {
```

Comment:
Do we have such method at RestUtils?

Reply:
We do not have data-import-utils dependency in pub-sub. This method is based on that implementation, but adjusted/simplified a little for pub-sub needs